### PR TITLE
feat(mysql/flexible): adding a mysql flexible managed instance

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3602

need https://github.com/jenkins-infra/azure-net/pull/132 from azure-net

### CODE REVIEW

File: mysql-public-db.tf [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+# Manages the MySQL Flexible Server https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server
+resource "azurerm_mysql_flexible_server" "public_db_mysql" {
+  name                         = "public-db-mysql"
+  resource_group_name          = data.azurerm_resource_group.public.name
+  location                     = var.location
+  administrator_login          = local.public_db_mysql_admin_login
+  administrator_password       = random_password.public_db_mysql_admin_password.result
+  sku_name                     = "B_Standard_B1ms" # 1vCore / 2 Gb - https://docs.microsoft.com/en-us/azure/virtual-machines/sizes-b-series-burstable
+  version                      = "8.0.21"          # TODO can be 5.7 as per https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server
```
Comment (smerle33) [inline mysql-public-db.tf:52]: mysql 5.7 or 8.0.21 ?

File: mysql-public-db.tf [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  zone                         = "1"
+  geo_redundant_backup_enabled = false #TODO  Changing this forces a new MySQL Flexible Server to be created
```
Comment (smerle33) [inline mysql-public-db.tf:54]: do we need geo redundant backup ? if so, we need to set it up now to avoid recreating it

File: mysql-public-db.tf [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  private_dns_zone_id          = azurerm_private_dns_zone.public_db_mysql.id
+  delegated_subnet_id          = data.azurerm_subnet.public_db_vnet_mysql_tier.id
+
+  high_availability { # TODO  Changing this forces a new MySQL Flexible Server to be recreated
+    mode                      = "ZoneRedundant"
+    standby_availability_zone = "2"
+  }
```
Comment (smerle33) [inline mysql-public-db.tf:61]: do we need to provide a stockage size (like for postgres) ... need to dive in the azure documentation

File: mysql-public-db.tf [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# Reference to the PostgreSQL/MySql dedicated network external resources
+# ALREADY LOADED IN postgres-public-db.tf
+# data "azurerm_virtual_network" "public_db"
+
+# Dedicated subnet is reserved as "delegated" for the mysql server (same as the postgres) on the public network
+# not specificaly needed as per the mysql flexible server documentation :
+# Ref. https://learn.microsoft.com/en-us/azure/mysql/flexible-server/concepts-networking
+# but seem adequate to use Private access (VNet integration) https://learn.microsoft.com/en-us/azure/mysql/flexible-server/concepts-networking#choose-a-networking-option
```
Comment (smerle33) [inline mysql-public-db.tf:8]: I followed the postgres architecture but it seem not mandatory for mysql, WDYT ?

File: mysql-public-db.tf (hunk 1 above)
Comment (smerle33) [reply mysql-public-db.tf:52]: 8.0.21 : https://matomo.org/faq/on-premise/matomo-requirements/#recommended-configuration

File: mysql-public-db.tf [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  administrator_login          = local.public_db_mysql_admin_login
+  administrator_password       = random_password.public_db_mysql_admin_password.result
+  sku_name                     = "B_Standard_B1ms" # 1vCore / 2 Gb - https://docs.microsoft.com/en-us/azure/virtual-machines/sizes-b-series-burstable
+  version                      = "8.0.21"          # TODO can be 5.7 as per https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server
+  zone                         = "1"
+  geo_redundant_backup_enabled = false #TODO  Changing this forces a new MySQL Flexible Server to be created
+  private_dns_zone_id          = azurerm_private_dns_zone.public_db_mysql.id
+  delegated_subnet_id          = data.azurerm_subnet.public_db_vnet_mysql_tier.id
+
+  high_availability { # TODO  Changing this forces a new MySQL Flexible Server to be recreated
```
[the remark this answers is not in the document]
Comment (smerle33) [reply mysql-public-db.tf:58]: useless for that kind of service and regarding the provider

File: mysql-public-db.tf (hunk 2 above)
Comment (smerle33) [reply mysql-public-db.tf:54]: false good enough for now

File: mysql-public-db.tf (hunk 3 above)
Comment (smerle33) [reply mysql-public-db.tf:61]: to remove high availiability

File: mysql-public-db.tf [hunk 6]
```diff
@@ -0,0 +1,86 @@
+# Reference to the PostgreSQL/MySql dedicated network external resources
+# ALREADY LOADED IN postgres-public-db.tf
+# data "azurerm_virtual_network" "public_db"
+
+# Dedicated subnet is reserved as "delegated" for the mysql server (same as the postgres) on the public network
+# not specificaly needed as per the mysql flexible server documentation :
+# Ref. https://learn.microsoft.com/en-us/azure/mysql/flexible-server/concepts-networking
+# but seem adequate to use Private access (VNet integration) https://learn.microsoft.com/en-us/azure/mysql/flexible-server/concepts-networking#choose-a-networking-option
```
Comment (dduportal) [inline mysql-public-db.tf:8]:
```suggestion
```

File: mysql-public-db.tf [hunk 7]
```diff
@@ -0,0 +1,86 @@
+# Reference to the PostgreSQL/MySql dedicated network external resources
+# ALREADY LOADED IN postgres-public-db.tf
+# data "azurerm_virtual_network" "public_db"
```
Comment (dduportal) [inline mysql-public-db.tf:3]: WDYT moving the data.azurerm_virtual_network.public_db definition from postgres-public-db.tf to vnets.tf ?

**Review by dduportal — APPROVED** [review]
`Plan: 9 to add, 0 to change, 0 to destroy.` Let's roll!

**Pull request merged**